### PR TITLE
Fix leaking PDs from pd.go e2e test

### PR DESCRIPTION
**Pull request by cjcullen**

### CODE REVIEW

**Comment by cjcullen** [comment]
If you've run e2e's on GCE/GKE recently, you've probably noticed some extra PDs being left behind. That was my fault.

I added an explicit delete w/ retries to the end of each test instead of relying on the deferred cleanup to delete them for me. Forcibly detaching/deleting PDs can be racy, where I can't detach the PD because it isn't attached, and then I can't delete the PD because it is attached. So, my best bet for now is retries.

**Comment by cjcullen** [comment]
@zmerlynn Can you take a look?

File: test/e2e/pd.go [hunk 1]
```diff
@@ -87,11 +87,22 @@ var _ = Describe("PD", func() {
...
 		By("waiting up to 60 seconds for host1Pod to start running")
-		expectNoError(waitForPodRunning(c, host1Pod.Name, 60*time.Second), "host1Pod not running after 60 seconds")
+		expectNoError(waitForPodRunning(c, host1Pod.Name, 120*time.Second), "host1Pod not running after 60 seconds")
 
 		By("deleting host1Pod")
 		expectNoError(podClient.Delete(host1Pod.Name), "Failed to delete host1Pod")
 
+		By(fmt.Sprintf("deleting PD %q", diskName))
+		err = deletePD(diskName, testContext.gceConfig.Zone)
+		for i := 0; i < 10; i++ {
```
Comment (zmerlynn) [inline test/e2e/pd.go]: Can you flip the expiration loop over to this style? https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/kubectl.go#L149 Seems to fit fine.

File: test/e2e/pd.go [hunk 2]
```diff
@@ -130,16 +141,27 @@ var _ = Describe("PD", func() {
...
 
 		By("deleting host0ROPod")
 		expectNoError(podClient.Delete(host0ROPod.Name), "Failed to delete host0ROPod")
 
 		By("deleting host1ROPod")
 		expectNoError(podClient.Delete(host1ROPod.Name), "Failed to delete host1ROPod")
+
+		By(fmt.Sprintf("deleting PD %q", diskName))
+		err = deletePD(diskName, testContext.gceConfig.Zone)
+		for i := 0; i < 10; i++ {
```
Comment (zmerlynn) [inline test/e2e/pd.go]: Ditto

File: test/e2e/pd.go [hunk 3]
```diff
@@ -130,16 +141,27 @@ var _ = Describe("PD", func() {
...
+			Logf("Couldn't delete PD. Sleeping 2 seconds")
+			time.Sleep(2 * time.Second)
+			if err = deletePD(diskName, testContext.gceConfig.Zone); err == nil {
+				break
+			}
+		}
+		expectNoError(err, "Error deleting PD")
```
Comment (zmerlynn) [inline test/e2e/pd.go]: If this becomes a common pattern, we could probably write a generic retry-on-error that just took a func. You're not mutating anything here, you're just pounding it. And I see why (stupid GCE case where we don't care, but really don't want to leak the resource).

**Comment by zmerlynn** [comment]
LHTM except minor nit.

File: test/e2e/pd.go (hunk 1 above)
Comment (cjcullen) [inline test/e2e/pd.go]: Done.

File: test/e2e/pd.go (hunk 2 above)
Comment (cjcullen) [inline test/e2e/pd.go]: Done.

File: test/e2e/pd.go (hunk 3 above)
Comment (cjcullen) [inline test/e2e/pd.go]: I have a callWithTimeout in the other PR I'm working on. I'll put it in a util package and switch these calls over to it when (if) that PR gets in.

**Comment by cjcullen** [comment]
@zmerlynn PTAL

**Pull request merged**